praw/config.py

```diff
@@ -1,162 +1,41 @@
-        self.custom = dict(Config.CONFIG.items(site_name), **settings)
-
-        self.client_id = self.client_secret = self.oauth_url = None
-        self.reddit_url = self.refresh_token = self.redirect_uri = None
-        self.password = self.user_agent = self.username = None
-
-        self._initialize_attributes()
-
-    def _fetch(self, key):
-        value = self.custom[key]
-        del self.custom[key]
-        return value
-
-    def _fetch_default(self, key, default=None):
-        if key not in self.custom:
-            return default
-        return self._fetch(key)
-
-    def _fetch_or_not_set(self, key):
-        if key in self._settings:  # Passed in values have the highest priority
-            return self._fetch(key)
-
-        env_value = os.getenv("praw_{}".format(key))
-        ini_value = self._fetch_default(key)  # Needed to remove from custom
-
-        # Environment variables have higher priority than praw.ini settings
-        return env_value or ini_value or self.CONFIG_NOT_SET
-
-    def _initialize_attributes(self):
-        self._short_url = (
-            self._fetch_default("short_url") or self.CONFIG_NOT_SET
-        )
-        self.check_for_updates = self._config_boolean(
-            self._fetch_or_not_set("check_for_updates")
-        )
-        self.kinds = {
-            x: self._fetch("{}_kind".format(x))
-            for x in [
-                "comment",
-                "message",
-                "redditor",
-                "submission",
-                "subreddit",
-                "trophy",
-            ]
-        }
-
-        for attribute in (
-            "client_id",
-            "client_secret",
-            "redirect_uri",
-            "refresh_token",
-            "password",
-            "user_agent",
-            "username",
-        ):
-            setattr(self, attribute, self._fetch_or_not_set(attribute))
-
-        for required_attribute in (
-            "oauth_url",
-            "ratelimit_seconds",
-            "reddit_url",
-            "timeout",
-        ):
-            setattr(self, required_attribute, self._fetch(required_attribute))
-
-        for attribute, conversion in {
-            "ratelimit_seconds": int,
-            "timeout": int,
-        }.items():
-            try:
-                setattr(self, attribute, conversion(getattr(self, attribute)))
-            except ValueError:
-                raise ValueError(
-                    "An incorrect config type was given for option {}. The "
-                    "expected type is {}, but the given value is {}.".format(
-                        attribute,
-                        conversion.__name__,
-                        getattr(self, attribute),
-                    )
-                )
+from configparser import ConfigParser
+from typing import Dict, Sequence
+
+
+def get_praw_environment_settings() -> Dict[str, str]:
+    return {k[len("praw_"):]: v for k, v in environ.items() if k.startswith("praw_")}
```

Comment:
Why can't this just be
```suggestion
    return {k[5:]: v for k, v in environ.items() if k.startswith("praw_")}
```